src/selectors/transaction$.js

```diff
@@ -12,15 +14,92 @@ export const activeTransactions$ = createSelector(
     transactions$,
     hasContent$,
     activeAccounts$,
+    activeCashlinks$,
     filterUnclaimedCashlinks$,
-    (transactions, hasContent, accounts, filterUnclaimed) => hasContent && new Map([...transactions.entries()].filter(entry => {
-        const tx = entry[1];
-        if (!accounts.has(tx.sender) && !accounts.has(tx.recipient)) return false;
-        if (filterUnclaimed) {
-            if (tx.isCashlink !== 'funding') return false;
-            const cashlinkAccount = accounts.get(tx.recipient);
-            return cashlinkAccount && !cashlinkAccount.cashlinkClaimed;
-        }
-        return true;
-    }))
+    (transactions, hasContent, accounts, cashlinks, filterUnclaimed) => hasContent && new Map([...transactions.entries()]
+        .filter(entry => {
+            const tx = entry[1];
+            if (!accounts.has(tx.sender) && !accounts.has(tx.recipient) &&
+                !cashlinks.has(tx.sender) && !cashlinks.has(tx.recipient)) return false;
+            if (filterUnclaimed) {
+                if (tx.isCashlink !== 'funding') return false;
+                const cashlink = cashlinks.get(tx.recipient);
+                return cashlink && cashlink.status <= CashlinkStatus.UNCLAIMED;
+            }
+            return true;
+        }))
+);
+
+const typedTransactions$ = createSelector(
+    activeTransactions$,
+    activeAccounts$,
+    transactions$,
+    (txs, accounts, txStore) => {
+        if (!accounts) return txs;
+        return txs && new Map([...txs.entries()].map(entry => {
+            const tx = entry[1];
+            const sender = accounts.get(tx.sender);
+            const recipient = accounts.get(tx.recipient);
+
+            // 1. Detect tx type
+
+            if (tx.isCashlink === 'claiming' && !recipient) {
+                // This is the tx where the final recipient claimed our outgoing cashlink.
+                // It will be displayed as an info bar only.
+                tx.type = 'cashlink_remote_claim';
```

Comment:
This selector finds and assignes _two_ additional props for a transaction: 1) it's `type`, which changes depending on which account you see the transaction from (when sending tx/cashlinks between your own accounts), and 2) the `pairedTx`, which does not change once it's known (except when it's updated when changing from pending to mined).

I agree, that usually selectors should have no side-effects onto the source data, but for the `pairedTx` this is actually great, because then the whole searching-all-stored-tx stuff does not need to be repeated on every account-switch.

So yes, it's a two-edged sword, but I think in this case manipulating the source data is worth it to prevent additional work during account-switching.